### PR TITLE
Allow selecting based on curation by dxDao or kleros

### PR DESCRIPTION
**Pull request by mtsalenc**
Takes the submission status from the Omen Verified Markets TCR directly to verify a market. `klerosTCRregistered`is a field added in https://github.com/protofire/omen-subgraph/pull/40

~Pending:~
~- [ ] The checkmark icon on #837 is not available in figma. This PR is currently using a star as a placeholder.~
~- [ ] The validity warning in the market details view does not link anywhere. Where should it link?~
~- [ ] The filter currently is `Valid` or `All`. I do not see a path to implement `Invalid` filter without polluting the code or breaking patterns since we need 2 queries separate queries for this: one for `klerosTCRregistered == false` and one for `klerosTCRregistered == null`.~

### CODE REVIEW

**Comment by pimato** [comment]
Hey @mtsalenc !
Thanks for preparing the Kleros market curation integration. We just merged the DXdao curation and it would be great to see the Kleros curation integrated ASAP.

> The checkmark icon on #837 is not available in figma. This PR is currently using a star as a placeholder.

The checkmarks should only be visible in the market details view and not on the main market overview. Please see my [Figma designs](https://www.figma.com/file/GNoehJ64fqfptghsjYqvTJ/Playground-V2?node-id=3421%3A0) as reference. The verified icon can be downloaded here: [dxdao-verified.svg.zip](https://github.com/protofire/omen-exchange/files/5156736/dxdao-verified.svg.zip)

> The validity warning in the market details view does not link anywhere. Where should it link?

The validity warning should not link anywhere. The main purpose is to alert the user that he/she should not touch the market as it is not verified.

> The filter currently is Valid or All. I do not see a path to implement Invalid filter without polluting the code or breaking patterns since we need 2 queries separate queries for this: one for klerosTCRregistered == false and one for klerosTCRregistered == null

@mariano-aguero successfully integrated the market validity picker for valid and not valid. I hope this eliminates your issue.

**Comment by mtsalenc** [comment]
The point that is confusing me is this: #837 requests that we filter and display markets based on whether it they are curated or not. However, there seems to be 2 ways to tell whether a market is curated or not:

- A market is valid if it was curated by DxDao ([i.e. this line is executed](https://github.com/protofire/omen-subgraph/blob/21fd721bdfcd3ab25449f518e73957f00bd64431/src/DXTokenRegistryMapping.ts#L15)). This is the way things are implemented since #1052 was merged.
or
- A market is valid if it is verified on Omen Verified Markets ([i.e. when this line evaluates to true](https://github.com/protofire/omen-subgraph/blob/21fd721bdfcd3ab25449f518e73957f00bd64431/src/GeneralizedTCRMapping.ts#L66))

So which one is it? Or do we want to do something else entirely?

**Comment by pimato** [comment]
Hey @mtsalenc the goal is to load all markets from the DXdao on-chain curation **and** all markets from the Kleros on-chain curation. It make sense to have several on-chain registries for curations to make Omen more resiliant. Therefor we need to fetch all markets from all sources which have been considered as valid.

Please checkout my design playground to get a feeling for how it should work [here](https://www.figma.com/proto/GNoehJ64fqfptghsjYqvTJ/Playground-V2?node-id=3421%3A0&viewport=114%2C185%2C0.529635488986969&scaling=min-zoom)

**Comment by KadenZipfel** [comment]
Is it just me or is 'All Sources' failing to return any markets? If I switch to Rinkeby it seems to display the DxDao curated markets, but on Mainnet it doesn't return the Kleros curated markets, which indicates to me that it might be a subgraph issue. 

Additionally, we want to keep the advanced filters component open when selecting dropdown items, but that can be done in a separate issue.

File: app/src/components/market/common/advanced_filters/img/kleros.svg [hunk 1]
```diff
@@ -0,0 +1,51 @@
+
```
Comment (KadenZipfel) [inline app/src/components/market/common/advanced_filters/img/kleros.svg:1]: This should be a .tsx file so we can use it as a component for consistency.

File: app/src/components/market/sections/market_list/market_home.tsx [hunk 2]
```diff
@@ -223,13 +223,15 @@ export const MarketHome: React.FC<Props> = (props: Props) => {
   const [sortByDirection, setSortByDirection] = useState<'asc' | 'desc'>(currentFilter.sortByDirection)
   const [showSearch, setShowSearch] = useState<boolean>(currentFilter.title.length > 0 ? true : false)
   const [showAdvancedFilters, setShowAdvancedFilters] = useState<boolean>(
-    (currentFilter.currency || currentFilter.arbitrator || currentFilter.marketValidity === MarketValidity.INVALID) &&
+    (currentFilter.currency ||
+      currentFilter.arbitrator ||
+      currentFilter.curationSource === CurationSource.ALL_SOURCES) &&
```
Comment (KadenZipfel) [inline app/src/components/market/sections/market_list/market_home.tsx:228]: Should be `currentFilter.curationSource !== CurationSource.ALL_SOURCES`

File: app/src/components/market/sections/market_list/market_home.tsx [hunk 3]
```diff
@@ -311,7 +313,9 @@ export const MarketHome: React.FC<Props> = (props: Props) => {
 
   useEffect(() => {
     setShowAdvancedFilters(
-      (currentFilter.currency || currentFilter.arbitrator || currentFilter.marketValidity === MarketValidity.INVALID) &&
+      (currentFilter.currency ||
+        currentFilter.arbitrator ||
+        currentFilter.curationSource === CurationSource.ALL_SOURCES) &&
         !fetchMyMarkets,
     )
```
Comment (KadenZipfel) [inline app/src/components/market/sections/market_list/market_home.tsx:320]: We want the advanced filters to stay open when you select a dropdown item, but also not necessarily be open on page load. You can emulate the following code:

```
const isInitialMount = useRef(true)
useEffect(() => {
  if (isInitialMount.current) {
    isInitialMount.current = true
  } else {
    setShowAdvancedFilters(
      (currentFilter.currency || currentFilter.arbitrator || currentFilter.marketValidity) && !fetchMyMarkets,
    )
  }
}, [currentFilter, fetchMyMarkets])
```

File: app/src/queries/markets_home.ts [hunk 4]
```diff
@@ -16,9 +17,44 @@ export const MarketDataFragment = gql`
...
+  openingTimestamp: Date
+  arbitrator: string
+  category: string
+  templateId: number
+  scaledLiquidityParameter: number
+  klerosTCRregistered: boolean
+  curatedByDxDao: boolean
+  curatedByDxDaoOrKleros: boolean
+}
+
```
Comment (KadenZipfel) [inline app/src/queries/markets_home.ts:57]: These should just be imported from the types file

**Review by KadenZipfel — CHANGES_REQUESTED** [review]

**Comment by mtsalenc** [comment]
> Edit: It looks I've just a misunderstanding of the subgraph deployment flow. Looks like it has only been deployed to Rinkeby. I'll do a proper code review now, but it seems ok.

I've been using my own instance to test (mtsalenc/omen-subgraph-personal)

**Review by KadenZipfel — APPROVED** [review]

**Review by pimato — APPROVED** [review]
works great!

**Pull request merged**